homeassistant/components/eufylife_ble/sensor.py

```diff
@@ -0,0 +1,211 @@
+        EufyLifeRealTimeWeightSensorEntity(data),
+    ]
+
+    if data.client.supports_heart_rate:
+        entities.append(EufyLifeHeartRateSensorEntity(data))
+
+    async_add_entities(entities)
+
+
+class EufyLifeSensorEntity(SensorEntity):
+    """Representation of an EufyLife sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the weight sensor entity."""
+        self._data = data
+
+        self._attr_device_info = DeviceInfo(
+            name=MODEL_TO_NAME[data.model],
+            connections={(dr.CONNECTION_BLUETOOTH, data.address)},
+        )
+
+    @property
+    def available(self) -> bool:
+        """Determine if the entity is available."""
+        if self._data.client.advertisement_data_contains_state:
+            # If the device only uses advertisement data, just check if the address is present.
+            return async_address_present(self.hass, self._data.address)
+
+        # If the device needs an active connection, availability is based on whether it is connected.
+        return self._data.client.is_connected
+
+    @callback
+    def _handle_state_update(self, *args: Any) -> None:
+        """Handle state update."""
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Register callback."""
+        self.async_on_remove(
+            self._data.client.register_callback(self._handle_state_update)
+        )
+
+
+class EufyLifeRealTimeWeightSensorEntity(EufyLifeSensorEntity):
+    """Representation of an EufyLife real-time weight sensor."""
+
+    _attr_name = "Real-time weight"
+    _attr_native_unit_of_measurement = UnitOfMass.KILOGRAMS
+    _attr_device_class = SensorDeviceClass.WEIGHT
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the real-time weight sensor entity."""
+        super().__init__(data)
+        self._attr_unique_id = data.address + "_real_time_weight"
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the native value."""
+        if self._data.client.state is not None:
+            return self._data.client.state.weight_kg
+        return None
+
+    @property
+    def suggested_unit_of_measurement(self) -> str | None:
+        """Set the suggested unit based on the unit system."""
+        if self.hass.config.units is US_CUSTOMARY_SYSTEM:
+            return UnitOfMass.POUNDS
+
+        return UnitOfMass.KILOGRAMS
+
+
+class EufyLifeWeightSensorEntity(RestoreEntity, EufyLifeSensorEntity):
+    """Representation of an EufyLife weight sensor."""
+
+    _attr_name = "Weight"
+    _attr_native_unit_of_measurement = UnitOfMass.KILOGRAMS
+    _attr_device_class = SensorDeviceClass.WEIGHT
+
+    _weight_kg: float | None = None
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the weight sensor entity."""
+        super().__init__(data)
+        self._attr_unique_id = data.address + "_weight"
```

Comment:
```suggestion
        self._attr_unique_id = f"{data.address}_weight"
```

Please use `f-strings` where possible

Reply:
Fixed.